File: backend/src/main/java/com/example/hackathon/model/Task.java
package com.example.hackathon.model;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;

@Document(collection = "tasks")
public class Task {
    @Id
    private String id;
    
    private String studentId;
    private String title;
    private String description;
    private LocalDateTime startDateTime;
    private LocalDateTime endDateTime;
    private TaskStatus status = TaskStatus.PENDING;
    private TaskPriority priority = TaskPriority.MEDIUM;
    private LocalDateTime createdAt = LocalDateTime.now();
    private LocalDateTime updatedAt = LocalDateTime.now();
    private LocalDateTime completedAt;

    public enum TaskStatus {
        PENDING, ONGOING, COMPLETED
    }

    public enum TaskPriority {
        LOW, MEDIUM, HIGH
    }

    // Constructors
    public Task() {}

    public Task(String studentId, String title, String description, 
                LocalDateTime startDateTime, LocalDateTime endDateTime) {
        this.studentId = studentId;
        this.title = title;
        this.description = description;
        this.startDateTime = startDateTime;
        this.endDateTime = endDateTime;
    }

    // Getters and Setters
    public String getId() { return id; }
    public void setId(String id) { this.id = id; }

    public String getStudentId() { return studentId; }
    public void setStudentId(String studentId) { this.studentId = studentId; }

    public String getTitle() { return title; }
    public void setTitle(String title) { this.title = title; }

    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }

    public LocalDateTime getStartDateTime() { return startDateTime; }
    public void setStartDateTime(LocalDateTime startDateTime) { this.startDateTime = startDateTime; }

    public LocalDateTime getEndDateTime() { return endDateTime; }
    public void setEndDateTime(LocalDateTime endDateTime) { this.endDateTime = endDateTime; }

    public TaskStatus getStatus() { return status; }
    public void setStatus(TaskStatus status) { 
        this.status = status;
        this.updatedAt = LocalDateTime.now();
        if (status == TaskStatus.COMPLETED && this.completedAt == null) {
            this.completedAt = LocalDateTime.now();
        }
    }

    public TaskPriority getPriority() { return priority; }
    public void setPriority(TaskPriority priority) { this.priority = priority; }

    public LocalDateTime getCreatedAt() { return createdAt; }
    public void setCreatedAt(LocalDateTime createdAt) { this.createdAt = createdAt; }

    public LocalDateTime getUpdatedAt() { return updatedAt; }
    public void setUpdatedAt(LocalDateTime updatedAt) { this.updatedAt = updatedAt; }

    public LocalDateTime getCompletedAt() { return completedAt; }
    public void setCompletedAt(LocalDateTime completedAt) { this.completedAt = completedAt; }
}

File: app/src/main/java/com/smarteval/app/data/model/Task.kt
package com.smarteval.app.data.model

import android.os.Parcelable
import kotlinx.parcelize.Parcelize

@Parcelize
data class Task(
    val id: String,
    val studentId: String,
    val title: String,
    val description: String?,
    val startDateTime: String,
    val endDateTime: String,
    val status: TaskStatus,
    val priority: TaskPriority,
    val createdAt: String,
    val updatedAt: String,
    val completedAt: String?,
    val isOverdue: Boolean
) : Parcelable

enum class TaskStatus {
    PENDING, ONGOING, COMPLETED
}

enum class TaskPriority {
    LOW, MEDIUM, HIGH
}

data class TaskStats(
    val pending: Int,
    val ongoing: Int,
    val completed: Int,
    val overdue: Int
)

data class CreateTaskRequest(
    val title: String,
    val description: String?,
    val startDateTime: String,
    val endDateTime: String,
    val priority: String
)